### PR TITLE
feat: bump jemalloc

### PR DESCRIPTION
**Pull request by dantengsky**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

bump tikv-jemallock to 0.5.2 (jemalloc to 5.3.0-patched)

Closes #issue

### CODE REVIEW

File: .github/workflows/dev-linux.yml [hunk 1]
```diff
@@ -52,6 +52,8 @@ jobs:
   build_musl:
     timeout-minutes: 30
     name: build_${{ matrix.arch }}_musl
+    env:
+      je_cv_pthread_getname_np: no
```
Comment (Xuanwo) [inline .github/workflows/dev-linux.yml:56]: How about adding a link to explain what does this mean?

Comment (dantengsky) [reply .github/workflows/dev-linux.yml]: @Xuanwo  it works & some docs have been added, PTAL.

**Comment by BohuTANG** [comment]
Test with hits:
```
#!/bin/bash

TRIES=3
QUERY_NUM=1

header="| query  | 1st(Cold) | 2nd(Hot) | 3rd(Hot) |\n| ------- | ------- | ------- | ------- |\n";
printf "$header"
cat queries.sql | while read query; do
    echo -n "| q${QUERY_NUM} | "
    for i in $(seq 1 $TRIES); do
        RES=$(curl -w 'Time: %{time_total}\n' http://default@localhost:8124?database=hits -d "${query}" 2>&1 | grep -P '^Time: ' | sed 's/Time: //')
        [[ "$?" == "0" ]] && echo -n "${RES}s" || echo -n "null"
        [[ "$i" != $TRIES ]] && echo -n " | "

        echo "${QUERY_NUM},${i},${RES}" >> result.csv
    done
    echo "|"

    QUERY_NUM=$((QUERY_NUM + 1))
done
```

./run.sh -- performance almost same as the main
| query  | 1st(Cold) | 2nd(Hot) | 3rd(Hot) |
| ------- | ------- | ------- | ------- |
| q1 | 0.151637s | 0.023416s | 0.022114s|
| q2 | 3.584200s | 0.348497s | 0.351163s|
| q3 | 0.741105s | 0.376482s | 0.400643s|
| q4 | 0.953572s | 0.398012s | 0.540593s|
| q5 | 4.463906s | 4.485443s | 4.459606s|
| q6 | 2.895282s | 2.745249s | 2.675295s|
| q7 | 0.614383s | 0.307875s | 0.355785s|
| q8 | 0.324200s | 0.468514s | 0.346739s|
| q9 | 1.371603s | 1.177860s | 1.193386s|
| q10 | 1.350739s | 1.327948s | 1.318463s|
| q11 | 1.284361s | 1.023517s | 0.956486s|
| q12 | 1.193605s | 1.170485s | 1.003818s|
| q13 | 1.082898s | 1.052376s | 1.051052s|
| q14 | 1.988733s | 1.932688s | 1.846677s|
| q15 | 1.467204s | 1.259660s | 1.272577s|
| q16 | 1.307992s | 1.387553s | 1.274356s|
| q17 | 3.329503s | 3.132690s | 3.144995s|
| q18 | 2.549586s | 2.747664s | 2.585862s|
| q19 | 6.500477s | 6.457675s | 6.506569s|
| q20 | 1.388967s | 0.085957s | 0.052583s|
| q21 | 4.733307s | 4.674647s | 4.620836s|
| q22 | 4.961573s | 4.902350s | 4.900670s|
| q23 | 9.853797s | 9.855365s | 9.829926s|
| q24 | 24.133292s | 23.273552s | 23.523848s|
| q25 | 1.317984s | 1.203797s | 1.113475s|
| q26 | 0.572090s | 0.621575s | 0.656769s|
| q27 | 1.269308s | 1.315654s | 1.303318s|
| q28 | 3.483059s | 3.524053s | 3.867667s|
| q29 | 6.024643s | 5.957900s | 5.929274s|
| q30 | 1.347715s | 1.257339s | 1.227176s|
| q31 | 1.707649s | 1.565176s | 1.694025s|
| q32 | 2.232822s | 2.405573s | 2.393751s|
| q33 | 12.639701s | 12.325156s | 12.562523s|
| q34 | 5.688450s | 5.620167s | 5.630758s|
| q35 | 9.266593s | 9.169079s | 9.165217s|
| q36 | 1.984606s | 1.981826s | 1.940012s|
| q37 | 0.803194s | 0.652162s | 0.651701s|
| q38 | 0.594475s | 0.410109s | 0.395452s|
| q39 | 0.528579s | 0.352259s | 0.416661s|
| q40 | 1.704771s | 1.699226s | 1.697364s|
| q41 | 0.358095s | 0.358831s | 0.327496s|
| q42 | 0.351379s | 0.217929s | 0.327619s|
| q43 | 0.337477s | 0.243061s | 0.216006s|

**Comment by PsiACE** [comment]
It may be necessary to test the build of the musl release, as it was previously possible to pass the development CI, but the production CI failed.

**Review by Xuanwo — APPROVED** [review]

**Comment by dantengsky** [comment]
> It may be necessary to test the build of the musl release, as it was previously possible to pass the development CI, but the production CI failed.

good point! any hints of how to test the musl production CI？

**Comment by Xuanwo** [comment]
> good point! any hints of how to test the musl production CI？

I believe the only different is production use release build for testing.

**Review by PsiACE — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**